docs/asciidoc/faq.adoc

```diff
@@ -157,6 +157,96 @@ myMethod.emptySequenceForKey("a") // this method returns empty Mono<String>
 ----
 ====
 
+[[faq.monoZipEmptyCompletion]]
+== Using `zip` along with empty-completed publishers
+
+When using the `zip` operator along with empty-completed publishers, it is important to be aware of the following behavior.
+
+Consider the following test case:
+
+====
+[source,java]
+----
+    @Test
+    public void testZipEmptyCompletionAllSubscribed() {
+        AtomicInteger cnt = new AtomicInteger();
+        Mono<Integer> mono1 = Mono.create(sink -> {
+            cnt.incrementAndGet();
+            sink.success();
+        });
+        Mono<Integer> mono2 = Mono.create(sink -> {
+            cnt.incrementAndGet();
+            sink.success();
+        });
+        Mono<Integer> zippedMono = Mono.zip(mono1, mono2, (v1, v2) -> v1);
+        zippedMono.subscribe();
+        assertEquals(2, cnt.get());
+    }
+----
+====
+
+While in this case the resulting `zippedMono` does not complete immediately upon empty completion of `mono1`, such behaviour is not guaranteed for all cases. For instance, consider the following test case:
```

Comment:
```suggestion
While in this case the resulting `zippedMono` subscribes to both `mono1` and `mono2`, such behaviour is not guaranteed for all cases. For instance, consider the following test case:
```

Reply:
I think the language mentioning completing immediately is not the essence here and can be confusing. Whether it completes before or after is an implementation detail, but the user is most probably interested in the act of subscribing and processing the sources.